src/passes/SeparateDataSegments.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Emits the data segments to a file. The file contains data from address base
+ * onwards (we must pass in base, as we can't tell it from the wasm - the
+ * first segment may start after a run of zeros, but we need those zeros in
+ * the file).
+ */
+#include "pass.h"
+#include "support/file.h"
+#include "wasm-features.h"
+#include "wasm.h"
+
+namespace wasm {
+
+struct SeparateDataSegments : public Pass {
+  bool modifiesBinaryenIR() override { return false; }
+
+  void run(Module* module) override {
+    std::string outfileName =
+      getPassOptions().getArgument("separate-data-segments",
+                                   "SeparateDataSegments usage: wasm-opt "
+                                   "--separate-data-segments@FILENAME");
+    Output outfile(outfileName, Flags::Binary);
+    std::string baseStr =
+      getPassOptions().getArgument("global-base",
+                                   "SeparateDataSegments usage: wasm-opt "
+                                   "--global-base=NUMBER");
```

Comment:
If this argument is only going to be used here, please prefix it, so `--separate-data-segments-global-base`. That's the convention we use in e.g. Asyncify which has several pass arguments.

Reply:
That seems odd.. but will do for conformance sake.

Reply:
Done.   Quite a mouthful but I guess that doesn't really matter since its all internal usage.